### ISSUE TITLE
Forum spam on turnkeylinux.org

### ISSUE THREAD

**Issue by bkahlerventer** [turnkeylinux/tracker#1796]
Forums on website are filled with spam.  if the site security is compromized, are the container images safe?

**Comment by JedMeister** [comment]
Hi @bkahlerventer - Thanks for your post.

Firstly, I'm curious why would you think that the website is insecure because we are targeted by spammers?

We have a range of anti-spam measures that stops most, if not all the spam bots. The current spammers (that are posting all the spam) are human and use email accounts provided by popular free services (e.g. gmail, outlook, proton, etc). So until they post, they look _exactly_ like a new user!

They also appear to be using VPNs to connect - so blocking by IP is not practical (or useful).

Obviously we could stop the spam, but the side effect would also block a genuine new user from being able to post.

So it's a constant balancing act between allowing new users to easily post, and blocking as much spam as we can (if you think it's bad at the moment, you should see the back end logs noting how much gets blocked!).

Another part of the issue is our website has pretty good SEO, which makes it a "high value" target for spammers.

We did have one other anti-spam module on the site that was fairly effective at blocking a lot of this sort of spam, but it also made lots of false positives. But middle of last year, that module started causing a huge amount of issues (e.g. https://github.com/turnkeylinux/tracker/issues/1742), so we disabled it.

So whilst I clean spam at least twice most days (sometimes only once a day on the weekend) we are currently getting hit really hard (about 10 new users per day, spamming up to 50 post each).

I have considered writing something myself to block/clean up the spam, but saying yes to one thing is always saying no to all the other things you could be doing. And at this point, whilst it does take a little bit of time to manually clean up the spam, it's fairly quick and easy and I have other, more important IMO 

So bottom line, the current situation is far from ideal, but the only options I'm left with are; make it harder for new users to post, put up with the spam or reorganise my priorities to write something to stop the spam (and hope that my efforts don't also have an undue negative impact on the ability for new users to post).

Regardless, thanks for posting. I totally get that seeing the forums filled with spam would be off putting (historically my first thought when seeing lots of spam in a forum is wondering if the project is still active?). It had never occurred to me that some might assume that there was some security issue!

So thanks for enlightening me about that! I might have to rethink my current position. Perhaps it would be better to block all posts from new users by default and require manual publishing?

Does that all make sense? Do you have further specific security concerns that I can address?

PS I'm going to close this issue - but I'll keep an eye out for your response, so please feel free to post back.

**Comment by bkahlerventer** [comment]
I used to run a few websites and faced comment spam myself.  It ended by
tightening security to posting comments.  Not sure about the current site
but Joomla and Wordpress had even comments enabled on images even though it
could not display it.  I highlighted it merely as a concern, as I had to
find spam in my inbox from the forums drowing out any legit content.  I
unsubscribed from the forums...

**Comment by JedMeister** [comment]
Thanks for your follow up. TBH, I actually hadn't considered the negative impact on users with regard to notifications about the spam. TBH, I'm not sure why I didn't consider that before?! That's a super important point and if we want more people involved in TurnKey then filling their inboxes with spam isn't going to do it!

So I think I'm going to look into making it so new users have to wait for me to publish their content...

Thanks again for your input! :smile:

**Comment by bkahlerventer** [comment]
I need to state that I appreciate the response and did see the posts in the
forums deleted so the care to the site is good.

I do not know your specific spam metrics in the forums but...

On the sites I ran, "real users" were created with scripts.  After using
some captcha techniques that will require Tensorflow level code in the
scripts, I managed to blocked scripts from creating users.

I also had to disable all identifiers of software versions and names I used
such as web server names, versions, as well as content management system
identifiers and versions and also plugin versions and identifiers...

Can I suggest a plugin that rate users and posts for possible spam so it
can be used to aid in moderation... not necessarily blocking new users, but
rating them low.  Also intelligent rate limiting... if a user start posting
too fast for a real user, the posts could reliably be rated as spam.
